### PR TITLE
Add unix helpers

### PR DESCRIPTION
**Pull request by InfRandomness**
This creates unix helper(s)

### CODE REVIEW

File: src/helpers.rs [hunk 1]
```diff
@@ -895,3 +895,7 @@ impl std::fmt::Display for HexRange {
         write!(f, "[{:#x}..{:#x}]", self.0.start.bytes(), self.0.end().bytes())
     }
 }
+
```
Comment (InfRandomness) [inline src/helpers.rs:898]: Should I consider adding `assert_target_os_is_unix` as well ? 
If yes, that would take care of snippets like 
```rs
        let target_os = &this.tcx.sess.target.os;
        assert!(
            target_os == "linux" || target_os == "macos",
            "`getenv` is only available for the UNIX target family"
        );
```

File: src/helpers.rs [hunk 2]
```diff
@@ -895,3 +895,7 @@ impl std::fmt::Display for HexRange {
...
+pub fn target_os_is_unix(target_os: &str) -> bool {
+    matches!(target_os, "linux" | "macos")
```
Comment (bjorn3) [inline src/helpers.rs:900]: The target spec contains a `family` field which can contain the `unix` value.

Comment (InfRandomness) [reply src/helpers.rs:900]: https://github.com/rust-lang/miri/pull/2011/files#r889613049

File: src/helpers.rs [hunk 3]
```diff
@@ -895,3 +895,7 @@ impl std::fmt::Display for HexRange {
         write!(f, "[{:#x}..{:#x}]", self.0.start.bytes(), self.0.end().bytes())
     }
 }
+
+pub fn target_os_is_unix(target_os: &str) -> bool {
```
Comment (RalfJung) [inline src/helpers.rs:899]:
```suggestion
/// Helper function used inside the shims of foreign functions to check that
/// `target_os` is a supported UNIX OS.
pub fn target_os_is_unix(target_os: &str) -> bool {
```

File: src/helpers.rs (hunk 1 above)
Comment (RalfJung) [reply src/helpers.rs:898]: Yes please do. :)

File: src/shims/dlsym.rs [hunk 4]
```diff
@@ -18,7 +19,8 @@ impl Dlsym {
     pub fn from_str<'tcx>(name: &[u8], target_os: &str) -> InterpResult<'tcx, Option<Dlsym>> {
         let name = &*String::from_utf8_lossy(name);
         Ok(match target_os {
-            "linux" | "macos" => unix::Dlsym::from_str(name, target_os)?.map(Dlsym::Posix),
+            target if target_os_is_unix(target) =>
+                unix::Dlsym::from_str(name, target_os)?.map(Dlsym::Posix),
```
Comment (RalfJung) [inline src/shims/dlsym.rs:23]:
```suggestion
                unix::Dlsym::from_str(name, target)?.map(Dlsym::Posix),
```
Let's be consistent about the binder we use.

**Comment by RalfJung** [comment]
This looks good! However, please rebase the PR on top of current master and also use assert_target_os_is_unix for getpid.

**Pull request merged**